sympy/printing/pycode.py

```diff
@@ -610,6 +612,109 @@ def _print_CodegenArrayPermuteDims(self, expr):
     def _print_CodegenArrayElementwiseAdd(self, expr):
         return self._expand_fold_binary_op('numpy.add', expr.args)
 
+    def _print_Indexed(self, expr):
+        elem = []
+        for i in range(expr.rank):
+            e = expr.indices[i]
+            if isinstance(e, Add) and e.has(Idx):
+                lower = 0
+                upper = 0
+                for a in e.args:
+                    if isinstance(a, Idx):
+                        lower += a.lower
+                        upper += a.upper
+                    else:
+                        lower += a
+                        upper += a
+                elem.append("%s:%s"%(lower, upper))
+            else:
+                elem.append(self._print(expr.indices[i]))
+        return "%s[%s]" % (self._print(expr.base.label), ', '.join(elem))
+
+    def _print_Idx(self, expr):
+        return "%s:%s"%(expr.lower, expr.upper)
+
+    def _print_For(self, expr):
```

Comment:
same here, `_print_For` should print a functional for loop should it not? (I don't see `for ... in ...:` being printed?)

Reply:
It's not `PythonCodePrinter`but `NumPyCodePrinter`you read. And for the `For` loop you want the vectorized form of NumPy. This is why you don't have `for ... in ...`

Reply:
I see, we should probably add independet tests to `test_pycode.py`.

Reply:
This is planned. Sorry to not push yet the test cases. But I want them to be clean.
I also try to keep the previous test unchanged as you wanted. I just have one to fix.